pkg/build/builder/dockerutil.go

```diff
@@ -156,15 +156,27 @@ func dockerRun(client DockerClient, createOpts docker.CreateContainerOptions, lo
 
 	containerName := containerNameOrID(c)
 
-	// Container was created, so we defer its removal.
-	defer func() {
+	// Container was created, so we defer its removal, and also remove it if we get a sigterm
```

Comment:
Please use Kube pkg/util/interrupt handlers for this, specifically Run(func() error)